Automattic/wp-calypso
# Hosting Command Palette: Display message when user doesn't have sites

Fixes https://github.com/Automattic/dotcom-forge/issues/4669

## Proposed Changes

Displays a message when the user doesn't have any sites:

![CleanShot 2023-12-20 at 04 32 23@2x](https://github.com/Automattic/wp-calypso/assets/36432/bc171121-b1a9-466b-91e7-707e1681e3a0)

When you have a site but none of your sites support a given feature, you'll see a more tailored message:

![CleanShot 2023-12-20 at 04 32 55@2x](https://github.com/Automattic/wp-calypso/assets/36432/4d4e0b63-5d11-4b8e-9684-72ed9d97966d)

![CleanShot 2023-12-20 at 04 33 22@2x](https://github.com/Automattic/wp-calypso/assets/36432/d76aee9b-4da6-49ba-a466-aaab3a3c7b87)

This is an improvement upon the current behavior, which simply shows an empty list:

![CleanShot 2023-12-18 at 16 25 34@2x](https://github.com/Automattic/wp-calypso/assets/36432/3bf77943-dabd-45b4-a144-a6cbe2a8885d)

I looked into including an "Add new site" link/button but: 1) there isn't really an existing UI pattern for this, and 2) it was hard/not intuitive to navigate to with the keyboard. I think we should bring it up with https://github.com/Automattic/dotcom-forge/issues/4630

## Testing Instructions

1. Create a new WordPress.com user and eject out of the signup flow before creating a site.
2. Visit the Sites page.
3. Open the command palette and verify "You don't have any sites yet." is displayed when you select "Manage hosting configuration".
4. With your existing WordPress.com user, verify the site list appears as expected because you have sites.
